### PR TITLE
asynchronous codes added and fixed

### PR DESCRIPTION
**Pull request by SouvikGhosh05**
### Describe your change:

* [x] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

This changes removes many bugs or typo of some existing code. Automated tests succeeded. I refactored most of them to look those codes nicer and faster. I also created a new directory of async features of Python so many Python developers can contribute to it.

Thank you. Regards~

### CODE REVIEW

File: asynchronous_programming/async_fibprime.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        return False
+
+    for i in range(2, int(sqrt(digit)) + 1):
+        if digit % i == 0:
+            prime_flag = 1
+            break
+    return prime_flag == 0
+
+
+async def main() -> None:
```
[the remark this answers is not in the document]
Comment (SouvikGhosh05) [reply asynchronous_programming/async_fibprime.py:48]: Doctest was added

[the remark this answers is not in the document]
Comment (SouvikGhosh05) [reply asynchronous_programming/async_fibprime.py:48]: Doctest was added for the main two functions.

**Comment by SouvikGhosh05** [comment]
@Kush1101 if you have any issues with this PR, kindly let me know thank you.

**Comment by cclauss** [comment]
Too many unrelated algorithms touched in a single with unrelated changes, some of which are not helpful.  `Asyncio` is for input/output operations and will actually slow down calculation operations so we do not want to feature that.

**Comment by SouvikGhosh05** [comment]
@cclauss  I actually refactored the codes with sourcery and retested with black formatter. That's why you can see too many file changes. I tried to include a separate asynchronous programs folder for all non-blocking code. Isn't this changes looks relevant to this repo?

**Pull request closed without merging**